great_expectations/data_context/types/base.py

```diff
@@ -960,7 +960,7 @@ class Meta:
         required=False,
         allow_none=True,
     )
-    id_ = fields.String(
+    id = fields.String(
         required=False,
         allow_none=True,
         data_key="id",
```

Comment:
Should be able to drop the `data_key` setting.

Reply:
Good call, done!